agent/s3/factory/factory.go

```diff
@@ -62,6 +64,19 @@ func (*s3ClientCreator) NewS3ClientForBucket(bucket, region string,
 	return s3manager.NewDownloaderWithClient(s3.New(sessWithRegion)), nil
 }
 
+// NewS3Client returns a new S3 client
+func (*s3ClientCreator) NewS3Client(region string,
```

Comment:
why not use the existing `NewS3ClientForBucket`?

Reply:
it only have the download file option. I don't need it, since I am not downloading anything on the instance

Reply:
I see, can you please update/add a comment for this func and the above `NewS3ClientCreator()` to reflect this.

Reply:
added comments